testing/scripts/seldon_e2e_utils.py

```diff
@@ -42,6 +43,27 @@ def wait_for_shutdown(deploymentName, namespace):
         ret = run(f"kubectl get -n {namespace} deploy/{deploymentName}", shell=True)
 
 
+def get_deployment_names(seldonDeploymentName, namespace, attempts=20, sleep=5):
+    for _ in range(attempts):
+        ret = run(
+            f"kubectl get -n {namespace} seldondeployment {seldonDeploymentName} -o yaml",
+            shell=True,
+            capture_output=True,
+        )
+        if ret.returncode == 0:
+            logging.warning(
+                f"Successfully waited for SeldonDeployment {seldonDeploymentName}"
+            )
+            break
+        logging.warning(
+            f"Unsuccessful wait command but retrying for SeldonDeployment {seldonDeploymentName}"
+        )
+        time.sleep(sleep)
+    assert ret.returncode == 0
```

Comment:
It may be better to throw an exception with a description so it's more informative on error

Reply:
Again, was following structure already present in the code, see `wait_for_rollout` function.
Shall I change all assertions into exceptions then?

Reply:
I added more meaningful message but left this as assertion (AssertionError seems good enough for tests). Please, let me know if you prefer some other exception to be raised there.